### ISSUE TITLE
Clean up / unify message and event format

### ISSUE THREAD

**Issue by matejpavlovic** [filecoin-project/mir#129]
Mir is currently using Protobufs for internal events and messages, while the libp2p transport layer is wrapping the Protobufs in CBOR. The gRPC based transport, however, is using Protobufs directly. Events and messages are also defined in a rather ad-hoc manner. This could use some cleaning up.

**Comment by xosmig** [comment]
Is there anything blocking us from migrating completely from protobuf to CBOR?
Protobufs have been a pain to deal with anyway.

**Comment by matejpavlovic** [comment]
No, not really. Protobuf was used initially by the original implementation and I did not have a strong reason to change it. Especially in the start, they were not even that much of a pain to deal with, as they were only used for rather simple things.
I haven't studied the pros and cons of migrating to CBOR (don't even know whether CBOR is significantly easier to deal with).
